XiEditor/AppDelegate.swift

```diff
@@ -409,6 +409,57 @@ class AppDelegate: NSObject, NSApplicationDelegate, XiClient {
         }
     }
 
+    @IBAction func openDocument(_ sender: Any?) {
+        let dialog = NSOpenPanel()
+
+        dialog.allowsMultipleSelection = false
+        dialog.canChooseFiles = true
+        dialog.canCreateDirectories = true
+        dialog.canChooseDirectories = true
+
+        guard
+            dialog.runModal() == .OK,
+            let result = dialog.url
+        else { return }
+
+        var isDirectory : ObjCBool = false
```

Comment:
Just general formatting nit, typically type colons are placed next to the variable, so this should look like `var isDirectory: ObjcBool = false`.